### PR TITLE
VY-RES-462 - API for flagged project

### PR DESCRIPTION
**Pull request by lf-achyutpkl**
@lf-kailashraj  @anishmanandhar  @krishnatimilsina  
please review my code

### CODE REVIEW

File: src/main/java/com/lftechnology/vyaguta/resource/dao/impl/ProjectDaoImpl.java [hunk 1]
```diff
@@ -73,4 +73,26 @@ public ProjectDaoImpl() {
...
+        List<Map<String, Object>> output = new ArrayList<>();
+
+        Query query = em
+                .createQuery(
+                        "SELECT p.title, ps.title, MAX(c.endDate) FROM Contract c JOIN c.project p JOIN p.projectStatus ps  WHERE ps.title = :status AND c.endDate < :date GROUP BY p.id, ps.title")
+                .setParameter("status", projectStatus).setParameter("date", LocalDate.now());
+        List<Object[]> result = query.getResultList();
+        for (Object[] obj : result) {
+            output.add(new HashMap<String, Object>() {
+                {
```
Comment (bhuwang) [inline src/main/java/com/lftechnology/vyaguta/resource/dao/impl/ProjectDaoImpl.java]: MAJOR Move the contents of this initializer to a standard constructor or to field initializers. [rule](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AS1171)

File: src/main/java/com/lftechnology/vyaguta/resource/rs/ProjectRs.java [hunk 2]
```diff
@@ -24,6 +25,7 @@
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.UriInfo;
 
+import org.hibernate.validator.constraints.NotBlank;
```
Comment (bhuwang) [inline src/main/java/com/lftechnology/vyaguta/resource/rs/ProjectRs.java]: MINOR Remove this unused import 'org.hibernate.validator.constraints.NotBlank'. [rule](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by bhuwang** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by bhuwang** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/main/java/com/lftechnology/vyaguta/resource/dao/ProjectDao.java [hunk 3]
```diff
@@ -16,4 +16,6 @@
 public interface ProjectDao extends CrudDao<Project, UUID> {
 
     List<Map<String, Object>> findBookedResource(LocalDate date);
+
+    List<Map<String, Object>> findFlaggedProjects(String projectStatus);
```
Comment (lf-kailashraj) [inline src/main/java/com/lftechnology/vyaguta/resource/dao/ProjectDao.java]: java doc

**Pull request merged**